### ISSUE TITLE
Failing tests on Codeship.io

### ISSUE THREAD

**Issue by alukach** [cugos/dropchop#226]
Codeship.io seems to be throwing errors when running `./codeship/test.sh`:

``` bash
[05:33:30] Using gulpfile ~/src/github.com/cugos/dropchop/gulpfile.js
[05:33:30] Starting 'topojson'...
[05:33:30] Starting 'css_vendor'...
[05:33:30] Starting 'mapbox_assets'...
[05:33:30] Starting 'fa-fonts'...
[05:33:30] Starting 'lint'...
[05:33:30] Starting 'js_dropchop'...
[05:33:30] Starting 'html'...
[05:33:30] Starting 'sass'...
[05:33:30] Starting 'assets'...
[05:33:30] Finished 'css_vendor' after 154 ms
[05:33:30] Finished 'html' after 205 ms
/home/rof/src/github.com/cugos/dropchop/src/js/layerlist.dropchop.js: line 120, col 18, ['extent'] is better written in dot notation.

1 error
[05:33:30] Finished 'sass' after 297 ms
[05:33:31] Finished 'fa-fonts' after 740 ms
[05:33:31] Finished 'lint' after 816 ms
[05:33:31] Finished 'js_dropchop' after 1.29 s
[05:33:31] Starting 'js'...
[05:33:31] Finished 'js' after 8.02 s
[05:33:31] Finished 'mapbox_assets' after 1.35 s
[05:33:31] Finished 'assets' after 1.33 s
[05:33:31] Starting 'build'...
[05:33:31] Finished 'build' after 8.87 s
[05:33:32] Finished 'topojson' after 1.94 s
[05:33:32] Starting 'js_vendor'...
[05:34:02] Finished 'js_vendor' after 30 s
[05:34:02] Starting 'vendor'...
[05:34:02] Finished 'vendor' after 11 s
[05:34:02] Starting 'build:prod'...
[05:34:02] Finished 'build:prod' after 8.56 s
[05:34:04] Using gulpfile ~/src/github.com/cugos/dropchop/gulpfile.js
[05:34:05] Starting 'test'...
02 11 2015 05:34:05.190:INFO [karma]: Karma v0.13.10 server started at http://localhost:9876/
02 11 2015 05:34:05.195:INFO [launcher]: Starting browser PhantomJS
02 11 2015 05:34:07.866:INFO [PhantomJS 1.9.8 (Linux 0.0.0)]: Connected on socket 9CUaoKp5lFvLk9PTAAAA with id 414491

Start:
  Dropchop!
    dropchop.js
       Throws error without options set
       Options are set
       Throws error when dc element is not found
    notify.dropchop.js
       dc.notify() - appear & disappear
       notification closes when close button is clicked
    dropzone.dropchop.js
       dc.dropzone event listeners
    left menu
       order is accurate
  form.dropchop.js
     dc.form.create()
     dc.form.remove()
     dc.form.createParam()
    dc.form.inputs
       dc.form.inputs.number()
       dc.form.inputs.select()
       dc.form.inputs.checkbox()
       dc.form.inputs.switch()
       dc.form.inputs.recursive()
  layerlist.dropchop.js
     after adding a layer, the DOM element exists
  layers.dropchop.js
     dc.layers.add()
     dc.layers.remove()
     dc.layers.duplicate()
     dc.layers.rename()
     dc.layers.makeLayer()
  selection.dropchop.js
     dc.selection.add()
     dc.selection.remove()
     dc.selection.clear()
  util.dropchop.js
     dc.util.removeFileExtension()
     dc.util.removeWhiteSpace()
     dc.util.concat()
     dc.util.xhr()
     dc.util.readFile()
     dc.util.executeUrlParams()
     dc.util.getFileExtension()
     dc.util.getBBox()
     dc.util.uncollect()
     dc.util.executeFC()
     dc.util.loader()
     dc.util.jsonFromUrl()
     dc.util.updateSearch()
ERROR: Object{position: 0, totalSize: 0, lengthComputable: false, loaded: 0, total: 0, returnValue: true, timeStamp: 1446442448921, eventPhase: 2, target: Object{readyState: 4, response: '', responseXML: null, onload: function () { ... }, onerror: function (a) { ... }, onloadstart: null, status: 0, onabort: null, upload: Object{onabort: ..., onload: ..., onprogress: ..., onerror: ..., onloadstart: ...}, onreadystatechange: null, responseType: '', onprogress: null, withCredentials: false, responseText: '', statusText: ''}, defaultPrevented: false, srcElement: Object{readyState: 4, response: '', responseXML: null, onload: function () { ... }, onerror: function (a) { ... }, onloadstart: null, status: 0, onabort: null, upload: Object{onabort: ..., onload: ..., onprogress: ..., onerror: ..., onloadstart: ...}, onreadystatechange: null, responseType: '', onprogress: null, withCredentials: false, responseText: '', statusText: ''}, type: 'error', clipboardData: , cancelable: true, currentTarget: Object{readyState: 4, response: '', responseXML: null, onload: function () { ... }, onerror: function (a) { ... }, onloadstart: null, status: 0, onabort: null, upload: Object{onabort: ..., onload: ..., onprogress: ..., onerror: ..., onloadstart: ...}, onreadystatechange: null, responseType: '', onprogress: null, withCredentials: false, responseText: '', statusText: ''}, bubbles: false, cancelBubble: false}
ERROR: Object{position: 0, totalSize: 0, lengthComputable: false, loaded: 0, total: 0, returnValue: true, timeStamp: 1446442448939, eventPhase: 2, target: Object{readyState: 4, response: '', responseXML: null, onload: function () { ... }, onerror: function (a) { ... }, onloadstart: null, status: 0, onabort: null, upload: Object{onabort: ..., onload: ..., onprogress: ..., onerror: ..., onloadstart: ...}, onreadystatechange: null, responseType: '', onprogress: null, withCredentials: false, responseText: '', statusText: ''}, defaultPrevented: false, srcElement: Object{readyState: 4, response: '', responseXML: null, onload: function () { ... }, onerror: function (a) { ... }, onloadstart: null, status: 0, onabort: null, upload: Object{onabort: ..., onload: ..., onprogress: ..., onerror: ..., onloadstart: ...}, onreadystatechange: null, responseType: '', onprogress: null, withCredentials: false, responseText: '', statusText: ''}, type: 'error', clipboardData: undefined, cancelable: true, currentTarget: Object{readyState: 4, response: '', responseXML: null, onload: function () { ... }, onerror: function (a) { ... }, onloadstart: null, status: 0, onabort: null, upload: Object{onabort: ..., onload: ..., onprogress: ..., onerror: ..., onloadstart: ...}, onreadystatechange: null, responseType: '', onprogress: null, withCredentials: false, responseText: '', statusText: ''}, bubbles: false, cancelBubble: false}

Finished in 0.823 secs / 0.41 secs

SUMMARY:
 37 tests completed
PhantomJS 1.9.8 (Linux 0.0.0) ERROR
SyntaxError: Unable to parse JSON string
at /home/rof/src/github.com/cugos/dropchop/dist/static/js/dropchop.js:1

Finished in 1.649 secs / 0.41 secs

SUMMARY:
 37 tests completed
Unhandled rejection Error: Not running
at Server.close (net.js:1233:11)
at /home/rof/src/github.com/cugos/dropchop/node_modules/karma/lib/server.js:338:17
at tryCatcher (/home/rof/src/github.com/cugos/dropchop/node_modules/karma/node_modules/bluebird/js/main/util.js:26:23)
at Promise._settlePromiseFromHandler (/home/rof/src/github.com/cugos/dropchop/node_modules/karma/node_modules/bluebird/js/main/promise.js:507:31)
at Promise._settlePromiseAt (/home/rof/src/github.com/cugos/dropchop/node_modules/karma/node_modules/bluebird/js/main/promise.js:581:18)
at Promise._settlePromises (/home/rof/src/github.com/cugos/dropchop/node_modules/karma/node_modules/bluebird/js/main/promise.js:697:14)
at Async._drainQueue (/home/rof/src/github.com/cugos/dropchop/node_modules/karma/node_modules/bluebird/js/main/async.js:123:16)
at Async._drainQueues (/home/rof/src/github.com/cugos/dropchop/node_modules/karma/node_modules/bluebird/js/main/async.js:133:10)
at Async.drainQueues (/home/rof/src/github.com/cugos/dropchop/node_modules/karma/node_modules/bluebird/js/main/async.js:15:14)
at process._tickCallback (node.js:415:13)
[05:34:08] Finished 'test' after 3.98 s
[05:34:11] 'test' errored after 6.98 s
[05:34:11] Error: 1
at formatError (/home/rof/src/github.com/cugos/dropchop/node_modules/gulp/bin/gulp.js:169:10)
at Gulp.<anonymous> (/home/rof/src/github.com/cugos/dropchop/node_modules/gulp/bin/gulp.js:195:15)
at Gulp.EventEmitter.emit (events.js:95:17)
at Gulp.Orchestrator._emitTaskDone (/home/rof/src/github.com/cugos/dropchop/node_modules/gulp/node_modules/orchestrator/index.js:264:8)
at /home/rof/src/github.com/cugos/dropchop/node_modules/gulp/node_modules/orchestrator/index.js:275:23
at finish (/home/rof/src/github.com/cugos/dropchop/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:21:8)
at cb (/home/rof/src/github.com/cugos/dropchop/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:29:3)
at removeAllListeners [as _onTimeout] (/home/rof/src/github.com/cugos/dropchop/node_modules/karma/lib/server.js:329:7)
at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)
```

This may be due to a lack of a requirement on Codeship.io? Tests run successfully on my machine.

**Comment by alukach** [comment]
Additional oddities mentioned in #191 and #224

**Comment by mapsam** [comment]
We are also getting these errors in Travis. Rerunning builds seems to fix them, so I've just been doing that for people's PR's - but that's not sustainable at all!

**Comment by mapsam** [comment]
Fixed in #257
